### ISSUE TITLE
the "await" for authorize and unlink is not waiting for the api

### ISSUE THREAD

**Issue by moroumo** [mix1009/FlutterDropbox#11]
Hi, 
Thank you for implementing this lib.

Not sure if this my code issue, setup issue or the lib issue.
In my test, the `await Dropbox.authorize()` is able to open a web browser for dropbox auth, but the function is returned immediately (not finish the web auth yet) and the `accessToken` would be always null.
the `authorize()` should wait for the web auth finishing.
```
      await Dropbox.authorize();
      accessToken = await Dropbox.getAccessToken();
```

Any suggestions?

thank you

**Comment by avdept** [comment]
@moroumo did you find any solution for this?

**Comment by SunilKividor** [comment]
@c3qo @avdept did u find any solution ?

**Comment by avdept** [comment]
> @c3qo @avdept did u find any solution ?

No, we abandoned dropbox in favour of using google cloud storage

**Comment by SunilKividor** [comment]
do u have any package in mind to implement dropbox in flutter ?

**Comment by c3qo** [comment]
> @c3qo @avdept did u find any solution ?

no. not support Dropbox any more😅

**Comment by vsaase** [comment]
@c3qo if found the bug at least for the unlink call: a missing `result` in `DropboxPlugin.m`, should be 
```
else if ([@"unlink" isEqualToString:call.method]) {
      [DBClientsManager unlinkAndResetClients];
      result(@(TRUE));
}
```

**Comment by SunilKividor** [comment]
@vsaase can you give me the location of the code where I need to change please. Thankyou

**Comment by SunilKividor** [comment]
@c3qo Did u find any solution for the authorize function not waiting ?

**Comment by mlava** [comment]
This is still an issue. It seems none of the authorisation calls are working asynchronously, only the upload/download/account ones work async.

Has anyone got a working async setup cobbled together?
